### DISCUSSION TITLE
Feature selection and survival analysis in a pipe

### DISCUSSION THREAD

**Discussion by DrChenziyan** [sebp/scikit-survival#326] [Q&A]
Hi, there, in my projects, I have more than 1000 features of radiomics that are extracted from MRI images. And I intended to build a pipeline of feature selection and the cox model to predict the prognosis. How can I combine the feature selection algorithms and prediction algorithms by sklearn.pipeline? 

As we know, we often used the Pipeline and GrdiSearchCV for classification such as
```python
from sklearn.pipeline import Pipeline
from sklearn.model_selection import GridSearchCV
from sklearn.svm import LinearSVC
from sklearn.ensemble import ExtraTreesClassifier
pipe = Pipeline([('lsvc', SelectFromModel(LinearSVC())),
                      ('etc', ExtraTreesClassifier())])
```
So, Can I use the sksurv models in the `SelectFromModel()`?
Hoping for your reply.

**Comment by sebp** [comment]
Just passing an instance of `CoxPHSurvivalAnalysis` to `SelectFromModel` should work fine.

**Reply by DrChenziyan** [reply]
Hi sebp, I still got some errors.

Here are my codes:
```python
pipe = Pipeline([
    ('scalar', StandardScaler()),
    ('cox', SelectFromModel(CoxnetSurvivalAnalysis(l1_ratio=1.0,alpha_min_ratio=0.01))),
    ('rsf', RandomSurvivalForest())
])

Kfold = StratifiedKFold(n_splits=3, shuffle=True, random_state=seed)
param_gird = {
    'rsf__n_estimators':[50,100,200,500],
    'rsf__max_depth':[5, 8, 10, 15, 20, 25],
    'rsf__max_features': ['auto', 'sqrt', 'log2'],
}
gcv = GridSearchCV(pipe, param_gird,
                           cv=Kfold, 
                           error_score='raise')
gcv.fit(X_train, y_train)
```
where my `X_train.shape` is (56,1037) and `y_train.shape` is (56, )
```python
X_train.head(2)
```
original_shape_Elongation | original_shape_Flatness | original_shape_LeastAxisLength|...
-- | -- | -- | -- 
0.876363 | 0.794211 | 40.457167| ... 
0.895718 | 0.466241 | 29.140086|... 
```python
y_train
```
 array([(False,  24.), ( True,  60.), (False,  11.), (False,   0.),
       (False,   4.), (False,   9.), (False,   1.), ( True,  35.),
       ( True,  19.), ( True,   7.), (False,   0.), ( True,   3.),
       (False,  37.), ( True,  23.), ( True,  28.), (False,   1.),
       (False,  13.), ( True,   4.), ( True,  74.), (False,   2.),
       ( True,  18.), ( True,  51.), (False,  11.), ( True,  59.),
       (False,   0.), (False,   5.), (False,  16.), (False,  16.),
       ( True,  21.), (False,   3.), ( True,   5.), ( True,  82.),
       (False,   8.), (False,   3.), ( True,  84.), (False,  21.),
       ( True, 133.), (False,   2.), (False,  17.), (False,   1.),
       (False,  10.), (False,  17.), (False,   1.), (False,   3.),
       ( True,  50.), (False,   2.), ( True,  44.), (False,  57.),
       (False,   1.), ( True,  23.), (False,  23.), (False,  15.),
       ( True,  41.), (False,   2.), (False,   3.), ( True,  71.)],
      dtype=[('Status', '?'), ('Survival_in_months', '<f8')])


However, I got the error with 
`ValueError: X has a different shape than during fitting.`

Could you give me some suggestions, Thank you~
